### PR TITLE
(PE-15719) Send puppet agent versions to dujour

### PR DESCRIPTION
**Pull request by smcclellan**
"puppet-agent-versions" is a new property which needs to be mutated to be
"puppet_agent_versions" before sending to dujour.

### CODE REVIEW

File: src/puppetlabs/dujour/version_check.clj [hunk 1]
```diff
@@ -163,7 +164,8 @@
                                 (log/warn e "Error occurred while checking for updates")
                                 (throw e)))]
         (if-not (nil? callback-fn)
-          (callback-fn server-response))))))
+          (callback-fn server-response)
+          server-response)))))
```
Comment (briancain) [inline src/puppetlabs/dujour/version_check.clj]: Was this change intentional? I guess if no callback function is defined it just tries to run version-check?

File: project.clj [hunk 2]
```diff
@@ -2,7 +2,7 @@
 (def tk-version "1.0.0")
 (def tk-jetty-version "1.0.0")
 
-(defproject puppetlabs/dujour-version-check "0.1.6-SNAPSHOT"
+(defproject puppetlabs/dujour-version-check "0.1.7-SNAPSHOT"
```
Comment (briancain) [inline project.clj]: I don't think you need to edit this right? I think jenkins will do it for us.

File: src/puppetlabs/dujour/version_check.clj [hunk 3]
```diff
@@ -81,7 +81,8 @@
         version-data {:version current-version}
         request-body (-> request-values
                          (dissoc :product-name)
-                         (set/rename-keys {:agent-os :agent_os})
+                         (set/rename-keys {:agent-os :agent_os
+                                           :puppet-agent-versions :puppet_agent_versions})
```
Comment (briancain) [inline src/puppetlabs/dujour/version_check.clj]: 👍

File: src/puppetlabs/dujour/version_check.clj (hunk 1 above)
Comment (smcclellan) [inline src/puppetlabs/dujour/version_check.clj]: Yep, without a callback function, it'll return the server's response as-is.

File: project.clj (hunk 2 above)
Comment (smcclellan) [inline project.clj]: Yeah I was doing this so I could test locally. Will revert.

File: src/puppetlabs/dujour/version_check.clj (hunk 3 above)
Comment (briancain) [inline src/puppetlabs/dujour/version_check.clj]: Actually, what happens if puppet-agent-versions isn't passed in to this version check? It seems like we get some weird behavior:

```clojure
user=> (def testmap {:agent_os "ohey"})
#'user/testmap
user=> (clojure.set/rename-keys {:agent_os :agent-os :puppet-agent-versions :puppet_agent_versions} testmap)
{:puppet-agent-versions :puppet_agent_versions, "ohey" :agent-os}
```

Comment (smcclellan) [inline src/puppetlabs/dujour/version_check.clj]: Aren't the arguments backwards there? The threading macro puts the `testmap` as the first argument.

Comment (briancain) [inline src/puppetlabs/dujour/version_check.clj]: Ah yep, you're right! Never mind.
